cns/restserver/ipam_test.go

```diff
@@ -287,166 +392,253 @@ func TestIPAMGetAlreadyAssignedIPConfigForSamePod(t *testing.T) {
+	for i := range ncIDs {
+		state, _ := NewPodStateWithOrchestratorContext(newPodIPs[i], newPodIPs[i], ncIDs[i], types.Assigned, prefixes[i], 0, testPod1Info)
+		desiredState[i] = state
+	}
 
-	desiredState, _ := NewPodStateWithOrchestratorContext(testIP1, testPod1GUID, testNCID, types.Assigned, 24, 0, testPod1Info)
+	firstAddress, _ := netip.ParseAddr(actualstate[0].IPAddress)
+	if firstAddress.Is4() == false && len(actualstate) > 1 {
+		actualstate[0], actualstate[1] = actualstate[1], actualstate[0]
+	}
+	for i := range actualstate {
+		assert.Equal(t, desiredState[i].GetState(), actualstate[i].GetState())
+		assert.Equal(t, desiredState[i].ID, actualstate[i].ID)
+		assert.Equal(t, desiredState[i].IPAddress, actualstate[i].IPAddress)
+		assert.Equal(t, desiredState[i].NCID, actualstate[i].NCID)
+		assert.Equal(t, desiredState[i].PodInfo, actualstate[i].PodInfo)
+	}
+}
+
+func TestIPAMAttemptToRequestIPNotFoundInPoolSingleNC(t *testing.T) {
+	ncIDs := []string{testNCID}
+	IPs := [][]string{{testIP1}, {testIP2}}
+	prefixes := []uint8{IPPrefixBitsv4}
+	IPAMAttemptToRequestIPNotFoundInPool(t, ncIDs, IPs, prefixes)
+}
 
-	assert.Equal(t, desiredState.GetState(), actualstate.GetState())
-	assert.Equal(t, desiredState.ID, actualstate.ID)
-	assert.Equal(t, desiredState.IPAddress, actualstate.IPAddress)
-	assert.Equal(t, desiredState.NCID, actualstate.NCID)
-	assert.Equal(t, desiredState.PodInfo, actualstate.PodInfo)
+func TestIPAMAttemptToRequestIPNotFoundInPoolMultipleNCs(t *testing.T) {
+	ncIDs := []string{testNCID, testNCIDv6}
+	IPs := [][]string{{testIP1, testIP1v6}, {testIP2, testIP2v6}}
+	prefixes := []uint8{IPPrefixBitsv4, IPPrefixBitsv6}
+	IPAMAttemptToRequestIPNotFoundInPool(t, ncIDs, IPs, prefixes)
 }
 
-func TestIPAMAttemptToRequestIPNotFoundInPool(t *testing.T) {
+func IPAMAttemptToRequestIPNotFoundInPool(t *testing.T, ncIDs []string, newPodIPs [][]string, prefixes []uint8) {
 	svc := getTestService()
 
 	// Add Available Pod IP to state
-	testState := NewPodState(testIP1, 24, testPod1GUID, testNCID, types.Available, 0)
-	ipconfigs := map[string]cns.IPConfigurationStatus{
-		testState.ID: testState,
-	}
-
-	err := UpdatePodIpConfigState(t, svc, ipconfigs)
-	if err != nil {
-		t.Fatalf("Expected to not fail adding IPs to state: %+v", err)
+	for i := range ncIDs {
+		ipconfigs := make(map[string]cns.IPConfigurationStatus, 0)
+		state := NewPodState(newPodIPs[0][i], prefixes[i], newPodIPs[0][i], ncIDs[i], types.Available, 0)
+		ipconfigs[state.ID] = state
+		err := UpdatePodIPConfigState(t, svc, ipconfigs, ncIDs[i])
+		if err != nil {
+			t.Fatalf("Expected to not fail adding IPs to state: %+v", err)
+		}
 	}
 
-	req := cns.IPConfigRequest{
+	req := cns.IPConfigsRequest{
 		PodInterfaceID:   testPod2Info.InterfaceID(),
 		InfraContainerID: testPod2Info.InfraContainerID(),
 	}
 	b, _ := testPod2Info.OrchestratorContext()
 	req.OrchestratorContext = b
-	req.DesiredIPAddress = testIP2
+	req.DesiredIPAddresses = newPodIPs[1]
```

Comment:
What do you mean completely different? In the original test it would add an IP to the pool and then request one that isn't in the pool and make sure that it can't be found. In this I am doing the same thing where newPodIPs[0] is in the pool and newPodIPs[1] is not. By requesting the IPs that not in the pool we expect an error.